beancount_import/source/ofx.py

```diff
@@ -964,6 +969,9 @@ def get_subaccount_cash(inv401ksource: Optional[str] = None) -> str:
                     units = round(units, quantity_round_digits)
                 if raw.unitprice is not None:
                     unitprice = normalize_fraction(raw.unitprice)
+                # For fidelity ofx, the unit of bond has been multiplied by 100
+                if self.org == "fidelity.com" and raw.uniqueid.startswith("912") and units % 100 == 0:
```

Comment:
I wonder if we make this more general and future-proof by detecting that a rescaling is required.
I ran into a similar/same issue a while ago, and my solution was to keep the units but rescale the price.
I detected the problem by noticing that `market value != units * price`.
Can you see if adopting my solution solves the problem?

```@@ -714,12 +806,23 @@ class ParsedOfxStatement(object):
                 time_str = find_child(invpos, 'dtpriceasof')
                 t = parse_ofx_time(time_str)
                 date = t.date()
+                units=find_child(invpos, 'units', D)
+                unitprice=find_child(invpos, 'unitprice', D)
+                mktval=find_child(invpos, 'mktval', D)
+                if mktval and mktval > 0:
+                    ratio = units*unitprice/mktval
+                    # these thresholds are arbitrary and could be tightened
+                    if (ratio < 0.2) or (ratio > 5):
+                        id_type = find_child(invpos, 'uniqueidtype')
+                        unique_id = find_child(invpos, 'uniqueid')
+                        unitprice = mktval / units if units > 0 else None
+                        print(
+                            f"[{id_type} {unique_id}]: Mismatch between UNITS * UNITPRICE = {units*unitprice:.2f} and  MKTVAL = {mktval:.2f}. Inferring price: {unitprice:.3f}")
                 raw_balance_entries.append(
                     RawBalanceEntry(
                         date=date,
                         uniqueid=find_child(invpos, 'uniqueid'),
                         units=find_child(invpos, 'units', D),
-                        unitprice=find_child(invpos, 'unitprice', D),
+                        unitprice=unitprice,
                         inv401ksource=find_child(invpos, 'inv401ksource'),
                         filename=filename))```

Reply:
Thanks, this looks good!